superflore/docker.py

```diff
@@ -36,8 +35,27 @@ def map_directory(self, host, container=None, mode='rw'):
     def add_bash_command(self, cmd):
         self.bash_cmds.append(cmd)
 
-    def build(self):
-        self.image = self.client.images.build(path=self.dockerfile_directory)
+    def clear_commands(self):
+        self.bash_cmds = list()
+
+    def build(self, dockerfile):
+        dockerfile_directory = os.path.dirname(dockerfile)
+        if not (os.path.isdir(dockerfile_directory) and
+                os.path.isfile('%s/Dockerfile' % dockerfile_directory)):
+            raise NoDockerfileSupplied(
+                'You must supply the location of the Dockerfile.'
+            )
+        self.image = self.client.images.build(path=dockerfile_directory)
+
+    def login(self):
+        # TODO(allenh1): add OAuth here, and fall back on user input
+        # if the OAuth doesn't exist (however one finds that).
+        user = getpass('Docker user:')
```

Comment:
Does this fallback if in non-interactive shells?

Reply:
Nope. No it does not. I'll use the environment variables I set on the CI if they are available, then do a `os.isatty(0)` to throw an exception if they aren't on a tty.